flink-runtime-web/web-dashboard/app/scripts/common/filters.coffee

```diff
@@ -87,3 +87,27 @@ angular.module('flinkApp')
 
 .filter "percentage", ->
   (number) -> (number * 100).toFixed(0) + '%'
+
+.filter "parseWatermark", ->
+  (value) ->
+    if value <= -9223372036854776000
+      return 'No Watermark'
+    else
+      return value
+
+.filter "lowWatermark", ->
+  (watermarks, nodeid) ->
+    lowWatermark = "None"
+    if watermarks != null && watermarks[nodeid] && watermarks[nodeid].length
+      values = (watermark.value for watermark in watermarks[nodeid])
+      lowWatermark = Math.min.apply(null, values)
+      if lowWatermark <= -9223372036854776000
+        lowWatermark = "No Watermark"
```

Comment:
I think this can be confusing to users, because with the logic below it sometimes says `No watermark` or `None` depending on whether the metric was available or not. I think we should stick to one of the two. The comment about the constant applies here as well.